src/activities/Elsa.Activities.Http/Formatters/XMLContentFormatter.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Serialization;
+using Elsa.Activities.Http.Services;
+
+
+namespace Elsa.Activities.Http.Formatters
+{
+    public static class XMLContentFormatterCache
+    {
+        public static List<Type> TypeCache { get; set; }
+    }
+
+
+    public class XMLContentFormatter : IContentFormatter
+    {
+        public int Priority => 1;
+        public IEnumerable<string> SupportedContentTypes => new[] { "application/xml", "text/xml" };
+
+        public Task<object> ParseAsync(byte[] content, string contentType)
+        {
+
+            GetTypeCache();
+
+            string rootNodeTypeName = GetRootTypeName(ref content);
+
+            if (rootNodeTypeName != "")
+            {
+                Type rootType = GetRootType(rootNodeTypeName);
+                XmlSerializer xmlSerializer = new XmlSerializer(rootType);
+                var model = xmlSerializer.Deserialize(new MemoryStream(content));
+                return Task.FromResult<object>(model);
+            }
+
+            // this is not clear - what will be returned if theres no deserialization?
+            return null;
+        }
+
+
+        public Task<byte[]> ParseAsync(object content, string contentType)
+        {
+            Type rootType = content.GetType();
+            XmlSerializer xmlSerializer = new XmlSerializer(rootType);
+            using (MemoryStream stream = new MemoryStream())
+            {
+                xmlSerializer.Serialize(stream, content);
+                return Task.FromResult<byte[]>(stream.ToArray());
+            }
+        }
+
+        private static Type GetRootType(string rootNodeName)
+        {
+            return XMLContentFormatterCache.TypeCache.Where(t => t.Name == rootNodeName).FirstOrDefault();
```

Comment:
I'm very skeptical we need all the assembly loading logic. It seems overkill just to find a single type to match against the XML root element. I understand why you do it, but what if the XML element contains a root node whose corresponding type lives in an assembly not included in the type cache because it's filtered by assemblies starting with the name "Elsa"?

Reply:
We need some assembly loading, because GetType() didnt get me the requested type (think because its not used somewhere). So imho we need to load the dll. The only thing we should remove is the lazy loading. I think a ModelType and a binary path maybe enough, so we just load the named binary (if needed) and pass the type to the deserializer